WIP: Requested changes from pull request 941 implemented

ritzratz opened the pull request:
- Parameters C1 to C6 are fixed for a pitch and only calculated once now
- Constant parameters now defined as "const float"
- Use of cmath instead of math.h
- Clang format applied
- Doxygen comments added
- Updated testTPCParameters.cxx 
- Various changes (naming convention, n++ -> ++n, double -> floats, ...)

wiechula commented:
Hi @ritzratz, is there a reason why you closed PR #941 and opened a new one? You could have simply pushed your changes to the same branch ...

ritzratz commented:
Sorry @wiechula .. there is no specific reason. I simpliy didn't know. OK I will try this rebase stuff ...what exactly do I have to do?

wiechula commented:
`git fetch --all`
`git rebase origin/dev`
Then, there could be merge conflicts which you need to solve. Edit the corresponding files and remove the conflicts. Then
`git add <modified file>`
`git rebase continue`
if the rebase finished then check it compiles and everything works as expected, the
`git push -f ritzratz dev-GEMModel`

wiechula commented:
Hi @ritzratz,
is there any new on this? Please let me know in case you need help, then we can do it interactively in a vidyo session.

ritzratz commented:
Hi @wiechula ... I did all the steps you recently mentioned and I solved the merge conflicts. This worked fine. However compiling did not work anymore. As you suggested I think this was caused as some dependencies have been added (the errors were not caused by my source files) and I updated alidist, tried to compile again but then there were even more errors and I gave up. Thereupon I removed the hole framework and set up a fresh one right from the beginning. This worked and I could compile it.

The status now is: I have a fresh alidist (ok almost 20 days ago) with O2 which compiled fine on my local machine. My changes are on my fork at github which I think is outdated. So I somehow have to merge the files from my (outdated) fork to my local (fresh and working) alidist and than kind of do a pull request again ...? 

Maybe an interactive vidyo session would help. Sorry for this ...

alibuild commented:
Error while checking build/O2/o2-dev-fairroot for d84ee4360686618e79ead470f879488e9af640a0:
```
sw/BUILD/O2-latest/log
  CMake did not find one.

  Could not find a package configuration file provided by "DDS" with any of
  the following names:

    DDSConfig.cmake
    dds-config.cmake

  Add the installation prefix of "DDS" to CMAKE_PREFIX_PATH or set "DDS_DIR"
  to a directory containing one of the above files.  If "DDS" provides a
  separate development package or SDK, be sure it has been installed.
Call Stack (most recent call first):
  CMakeLists.txt:65 (include)


-- Boost version: 1.66.0
-- Found the following Boost libraries:
--   thread
--   system
--   timer
--   program_options
--   random
--   filesystem
--   chrono
--   exception
--   regex
--   serialization
--   log
--   log_setup
--   unit_test_framework
--   date_time
--   signals
--   atomic
-- Setting FairRoot environment…
-- FairRoot ... - found /mnt/mesos/sandbox/sandbox/sw/slc7_x86-64/FairRoot/dev-1
-- FairRoot Library directory  :     /mnt/mesos/sandbox/sandbox/sw/slc7_x86-64/FairRoot/dev-1/lib
-- FairRoot Include path…      :     /mnt/mesos/sandbox/sandbox/sw/slc7_x86-64/FairRoot/dev-1/include
-- FairRoot Cmake Modules      :     /mnt/mesos/sandbox/sandbox/sw/slc7_x86-64/FairRoot/dev-1/share/fairbase/cmake
-- FairRoot ... - found /mnt/mesos/sandbox/sandbox/sw/slc7_x86-64/FairRoot/dev-1
-- Looking for FairMQ functionality in FairRoot ...
CMake Error at cmake/modules/FindFairMQ.cmake:46 (message):
  FairRoot is not built with FairMQ support
Call Stack (most recent call first):
  cmake/O2Dependencies.cmake:59 (find_package)
  CMakeLists.txt:65 (include)


-- Configuring incomplete, errors occurred!
See also "/mnt/mesos/sandbox/sandbox/sw/BUILD/da83bbb27b96e6f50d85d24a94a256ceb97a5533/O2/CMakeFiles/CMakeOutput.log".
```
Full log [here](https://ali-ci.cern.ch/repo/logs/AliceO2Group/AliceO2/1058/d84ee4360686618e79ead470f879488e9af640a0/build_O2_o2-dev-fairroot/fullLog.txt).

alibuild commented:
Error while checking build/o2checkcode/o2 for d84ee4360686618e79ead470f879488e9af640a0:
```
sw/BUILD/O2-latest/log
 2/72 Test #32: test_TPCSimulation_testTPCGEMAmplification .........***Failed    4.22 sec
Running 4 test cases...
GEM SETUP TOOK 1.05
/mnt/mesos/sandbox/sandbox/sw/SOURCES/O2/0_O2_DATAFLOW/0/Detectors/TPC/simulation/test/testTPCGEMAmplification.cxx(53): error: in "GEMamplification_test": difference{0.0980058} between energyResolution{13.2858696} and 12.1{12.1} exceeds 5%
GEM SETUP TOOK 0.41
GEM SETUP TOOK 0.37
GEM SETUP TOOK 0.39
```
Full log [here](https://ali-ci.cern.ch/repo/logs/AliceO2Group/AliceO2/1058/d84ee4360686618e79ead470f879488e9af640a0/build_o2checkcode_o2/fullLog.txt).

alibuild commented:
Error while checking build/O2/o2 for d84ee4360686618e79ead470f879488e9af640a0:
```
sw/BUILD/O2-latest/log
 38/157 Test #112: test_TPCSimulation_testTPCGEMAmplification ...........................***Failed    4.89 sec
Running 4 test cases...
GEM SETUP TOOK 1.21
/mnt/mesos/sandbox/sandbox/sw/SOURCES/O2/dev/0/Detectors/TPC/simulation/test/testTPCGEMAmplification.cxx(53): error: in "GEMamplification_test": difference{0.0980058} between energyResolution{13.2858696} and 12.1{12.1} exceeds 5%
GEM SETUP TOOK 0.51
GEM SETUP TOOK 0.52
GEM SETUP TOOK 0.41
```
Full log [here](https://ali-ci.cern.ch/repo/logs/AliceO2Group/AliceO2/1058/d84ee4360686618e79ead470f879488e9af640a0/build_O2_o2/fullLog.txt).

alibuild commented:
Error while checking build/O2/o2-dev-fairroot for 0564dd45015c5c61c068a7493070b91c703260b9:
```
sw/BUILD/O2-latest/log
  CMake did not find one.

  Could not find a package configuration file provided by "DDS" with any of
  the following names:

    DDSConfig.cmake
    dds-config.cmake

  Add the installation prefix of "DDS" to CMAKE_PREFIX_PATH or set "DDS_DIR"
  to a directory containing one of the above files.  If "DDS" provides a
  separate development package or SDK, be sure it has been installed.
Call Stack (most recent call first):
  CMakeLists.txt:65 (include)


-- Boost version: 1.66.0
-- Found the following Boost libraries:
--   thread
--   system
--   timer
--   program_options
--   random
--   filesystem
--   chrono
--   exception
--   regex
--   serialization
--   log
--   log_setup
--   unit_test_framework
--   date_time
--   signals
--   atomic
-- Setting FairRoot environment…
-- FairRoot ... - found /mnt/mesos/sandbox/sandbox/sw/slc7_x86-64/FairRoot/dev-1
-- FairRoot Library directory  :     /mnt/mesos/sandbox/sandbox/sw/slc7_x86-64/FairRoot/dev-1/lib
-- FairRoot Include path…      :     /mnt/mesos/sandbox/sandbox/sw/slc7_x86-64/FairRoot/dev-1/include
-- FairRoot Cmake Modules      :     /mnt/mesos/sandbox/sandbox/sw/slc7_x86-64/FairRoot/dev-1/share/fairbase/cmake
-- FairRoot ... - found /mnt/mesos/sandbox/sandbox/sw/slc7_x86-64/FairRoot/dev-1
-- Looking for FairMQ functionality in FairRoot ...
CMake Error at cmake/modules/FindFairMQ.cmake:46 (message):
  FairRoot is not built with FairMQ support
Call Stack (most recent call first):
  cmake/O2Dependencies.cmake:59 (find_package)
  CMakeLists.txt:65 (include)


-- Configuring incomplete, errors occurred!
See also "/mnt/mesos/sandbox/sandbox/sw/BUILD/da83bbb27b96e6f50d85d24a94a256ceb97a5533/O2/CMakeFiles/CMakeOutput.log".
```
Full log [here](https://ali-ci.cern.ch/repo/logs/AliceO2Group/AliceO2/1058/0564dd45015c5c61c068a7493070b91c703260b9/build_O2_o2-dev-fairroot/fullLog.txt).

alibuild commented:
Error while checking build/o2/macos for d84ee4360686618e79ead470f879488e9af640a0:
```
sw/BUILD/O2-latest/log
Error in <TFile::TFile>: file tpc_polyadist.root does not exist
Running 4 test cases...
GEM SETUP TOOK 15.4
/build/ci_checks/o2_checker_macos_o2_ci_0/sw/SOURCES/O2/1058/0/Detectors/TPC/simulation/test/testTPCGEMAmplification.cxx:53: error: in "GEMamplification_test": difference{0.0980058} between energyResolution{13.2858696} and 12.1{12.1} exceeds 5%
GEM SETUP TOOK 0.42
GEM SETUP TOOK 0.43
GEM SETUP TOOK 0.42
```
Full log [here](https://ali-ci.cern.ch/repo/logs/AliceO2Group/AliceO2/1058/d84ee4360686618e79ead470f879488e9af640a0/build_o2_macos/fullLog.txt).

alibuild commented:
Error while checking build/O2/o2 for 0564dd45015c5c61c068a7493070b91c703260b9:
```
sw/BUILD/O2-latest/log
[32001]: network-interface                               = default                                                                                             <string>         [default] 
[32001]: port-range-max                                  = 32000                                                                                               <int>            [default] 
[32001]: port-range-min                                  = 22000                                                                                               <int>            [default] 
[32001]: rate                                            = 0                                                                                                   <float>          [default] 
[32001]: session                                         = default                                                                                             <string>         [default] 
[32001]: severity                                        = debug                                                                                               <string>         [default] 
[32001]: shm-monitor                                     = 0                                                                                                   <bool>           [default] 
[32001]: shm-segment-size                                = 2000000000                                                                                          <std::size_t>    [default] 
[32001]: transport                                       = zeromq                                                                                              <string>         [default] 
[32001]: verbosity                                       = medium                                                                                              <string>         [default] 
[32001]: 
[32001]: [00:42:42][STATE] Entering INITIALIZING DEVICE state
[32001]: 
[32001]: [00:42:42][STATE] Entering DEVICE READY state
[32001]: [00:42:42][STATE] Entering INITIALIZING TASK state
[32001]: [00:42:42][STATE] Entering READY state
[32001]: [00:42:42][STATE] Entering RUNNING state
[32001]: [00:42:42][INFO] DEVICE: Running...
[32001]: [00:42:42][INFO] foo[0]: in: -nan (-nan MB) out: -nan (-nan MB)
[32001]: [00:42:42][INFO] from_producer_to_consumer[0]: in: -nan (-nan MB) out: -nan (-nan MB)
[INFO] Quitting
[INFO] QUIT_REQUESTED

[32002]: [00:42:42][INFO] Received device shutdown request (signal 15).
[32002]: [00:42:42][INFO] Waiting for graceful device shutdown. Hit Ctrl-C again to abort immediately.
[32001]: [00:42:42][INFO] Received device shutdown request (signal 15).
[32001]: [00:42:42][INFO] Waiting for graceful device shutdown. Hit Ctrl-C again to abort immediately.
[32002]: [00:42:42][STATE] Entering DEVICE READY state
[32002]: [00:42:42][STATE] Entering INITIALIZING TASK state
[32002]: [00:42:42][STATE] Entering READY state
[32002]: [00:42:42][STATE] Entering RUNNING state
[32002]: [00:42:42][INFO] DEVICE: Running...
[32002]: [00:42:42][STATE] Entering READY state
[32001]: [00:42:43][STATE] Entering READY state
[32001]: [00:42:43][STATE] Entering RESETTING TASK state
[32001]: [00:42:43][STATE] Entering DEVICE READY state
[32001]: [00:42:43][STATE] Entering RESETTING DEVICE state
[32001]: [00:42:43][STATE] Entering IDLE state
[32001]: [00:42:43][STATE] Entering EXITING state
[32001]: [00:42:43][STATE] Exiting FairMQ state machine
[32001]: [00:42:43][INFO] Process 32001 is exiting.


96% tests passed, 1 tests failed out of 26

Total Test time (real) =  68.12 sec

The following tests FAILED:
	156 - test_Framework_test_WorkflowOptions (Timeout)
```
Full log [here](https://ali-ci.cern.ch/repo/logs/AliceO2Group/AliceO2/1058/0564dd45015c5c61c068a7493070b91c703260b9/build_O2_o2/fullLog.txt).

sawenzel commented:
Any news on this PR?
